File: lesson_18/src/main/java/org/lesson18/Product.java
package org.lesson18;

import jakarta.persistence.*;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id")
    private UUID id;
    @Column(name = "title")
    private String title;
    @Column(name = "price")
    private BigDecimal price;

    public UUID getId() {
        return id;
    }

    public List<Customer> getCustomers() {
        return customers;
    }

    @ManyToMany
    @JoinTable(
            name = "purchase_details",
            joinColumns = @JoinColumn(name = "product_id"),
            inverseJoinColumns = @JoinColumn(name = "customer_id")
    )

    private List<Customer> customers;

    public void setTitle(String title) {
        this.title = title;
    }

    public void setPrice(BigDecimal price) {
        this.price = price;
    }

    public Product(String title, BigDecimal price) {
        super();
        this.title = title;
        this.price = price;
    }

    public Product() {
    }

    public String getTitle() {
        return title;
    }

    public BigDecimal getPrice() {
        return price;
    }
}

File: lesson_18/src/main/kotlin/org/lesson18kt/Product.kt
package org.lesson18kt

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany
import jakarta.persistence.Table
import java.math.BigDecimal
import java.util.UUID

@Entity
@Table(name = "product")
data class Product(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id")
    var id: UUID? = null,
    @Column(name = "title")
    var title: String? = null,
    @Column(name = "price")
    var price: BigDecimal? = null,

    @ManyToMany
    @JoinTable(
        name = "purchase_details",
        joinColumns = [JoinColumn(name = "product_id")],
        inverseJoinColumns = [JoinColumn(name = "customer_id")]
    )
    var customers: List<Customer>? = null,
)
